Updating sync

phg49389 opened the pull request:
This update started out as simply trying to add a small time delay between each block sync, then evolved into making it URL-compatible to allow the user to choose how long to wait, then add a way to sync one block, then be able to choose if you send an email to everyone at the end of the "sync all blocks" phase.

The biggest change to the code was the refactoring from one class that had every method URL-callable into two classes: the one that is URL-callable, and the other that actually does the syncs and work and stuff. Fixed a few typos, organized the imports, made the indentation standardized, etc., etc.

phg49389 committed:
Made a lot of changes

Grouped methods that don’t need to be URL accessible into a separate
object, implemented it so that you can sync all the blocks and choose
whether or not you want to send an email, you can choose how long to
wait between each block sync, you can choose to sync one individual
block, and I sorted the imports alphabetically.

phg49389 committed:
Removed test code so it actually does its job.

c-schwarze commented:
Hey Phil, in the future, if you could separate the super commit you made into smaller ones, that would be helpful :) For example, you made a list of things you updated within that commit. Next time, make each of those things in the list a separate commit.

For reference: https://guides.github.com/introduction/flow/

e-chapin commented:
It is also helpful because you can look at what you have changed recently on your local build when you run into issues. Stuff like that

e-chapin commented:
I made some [inlines notes in the commits tab](https://github.com/betheluniversity/programs/commit/f49329d559f512e0fd798dc0d6ca606379ca3e77), but a few other things
1. Breaks if you don't enter a value for 'Number of seconds to wait between each sync'
2. [You could make your template a little more versatile by reading this](https://pythonhosted.org/Flask-Classy/#url-for-art-thou-romeo)
3. Template file indentation maybe could be cleaned up a bit.
4. There a few other PEP8 things you could clean up once the other changes are made.

Other than that great job! Logic looks good and my test run was successful.

phg49389 committed:
Addressed Eric's feedback and AnnMarie's suggestion

phg49389 committed:
Addressed a few more Github comments

Moved the before_request to an init, moved the after into the Flask
Class, fixed some indentation in both templates, and set the default
value of sync_all to 0.25 seconds, allowing the top field to be blank
and it won’t throw an error.

phg49389 committed:
Forgot to update references of CascaseBlockProcessor

Changed from local session variable to in the class. Also, removed now
unnecessary JSONEncoder.

Commit:
add raven

Commit:
move self.data so its available everywhere

Commit:
add some todos

Commit:
check the path before loading the block from Cascade
